test/format/hs.c

```diff
@@ -83,8 +83,13 @@ hs_cursor(void *arg)
         /* Search to the last-known location. */
         if (!restart) {
             cursor->set_key(cursor, hs_btree_id, &key, hs_start_ts, hs_counter);
+
+            /*
+             * Limit expected errors because this is a diagnostic check (the WiredTiger API allows
+             * prepare-conflict, but that would be unexpected from the history store file).
+             */
             ret = cursor->search_near(cursor, &exact);
-            testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+            testutil_assert(ret == 0 || ret == WT_NOTFOUND || ret == WT_ROLLBACK);
```

Comment:
If `search_near` fails with `WT_ROLLBACK`, then the cursor will be positioned at the beginning.  If the chance of a rollback is more than miniscule, it's possible we'll never visit the end of the history file.  Maybe continue the loop if we detect rollback?

Reply:
BTW, the evergreen queue looks like it's quite backed up.

Reply:
I think we're safe: the chance of rollback isn't that large (anecdotally), but we run this test a lot, we'll get to the end.